cli/vscodeipc_test.go

```diff
@@ -0,0 +1,45 @@
+package cli_test
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/cli/clitest"
+	"github.com/coder/coder/testutil"
+)
+
+func TestVSCodeIPC(t *testing.T) {
+	t.Parallel()
+	// Ensures the vscodeipc command outputs it's running port!
+	// This signifies to the caller that it's ready to accept requests.
+	t.Run("PortOutputs", func(t *testing.T) {
+		t.Parallel()
+		client, workspace, _ := setupWorkspaceForAgent(t, nil)
+		cmd, _ := clitest.New(t, "vscodeipc", workspace.LatestBuild.Resources[0].Agents[0].ID.String(),
+			"--token", client.SessionToken(), "--url", client.URL.String())
+		var buf bytes.Buffer
+		cmd.SetOut(&buf)
+		ctx, cancelFunc := testutil.Context(t)
+		defer cancelFunc()
+		done := make(chan error)
```

Comment:
Require below may exit test early, leaving goroutine behind.

```suggestion
		done := make(chan error, 1)
```